### PR TITLE
Fix coverage session inside generated project

### LINKED ISSUE #61: Coverage not working
I cloned the project recently and the coverage was not working as the action did not find any file to push. I found a way to fix it (probably not ideal, see https://github.com/mmsg-warwick/encapsulated-LTES/pull/4).

### PR DESCRIPTION
**Pull request by santacodes**
The coverage session was the same as the fix Ferran mentioned in this [commit](https://github.com/pybamm-team/pybamm-cookie/commit/5afcceeccd157aa12177a53b834748da7c39e446), reverting to this.
Fixes #61

### CODE REVIEW

**Comment by santacodes** [comment]
Also would this fix require a new release of version `v0.1.2`?

File: template/{{ project_name }}/noxfile.py.jinja [hunk 1]
```diff
@@ -56,9 +56,10 @@ def run_user_tests(session):
 @nox.session(name="coverage")
 def run_coverage(session):
     """Run the coverage tests and generate an XML report."""
-    session.posargs.append("--cov=src/{{project_slug}}")
-    session.posargs.append("--cov-report=xml")
-    run_user_tests(session)
+    session.install("setuptools", silent=False)
```
Comment (agriyakhetarpal) [inline template/{{ project_name }}/noxfile.py.jinja:59]: Are we sure we need to install `setuptools` here? Does we receive failures without it?

**Review by agriyakhetarpal — APPROVED** [review]
Thanks, @santacodes! Yes, it would be nice to publish a v0.1.2 release.

File: template/{{ project_name }}/noxfile.py.jinja (hunk 1 above)
Comment (santacodes) [reply template/{{ project_name }}/noxfile.py.jinja:59]: Oops I didn't notice this, we certainly don't need setuptools. I'll remove it.

Comment (santacodes) [reply template/{{ project_name }}/noxfile.py.jinja:59]: @agriyakhetarpal I think we do need setuptools, what if the generated project has a setuptools backend?

Comment (agriyakhetarpal) [reply template/{{ project_name }}/noxfile.py.jinja:59]: I see, we can keep it in that case :)

**Pull request merged**